src/components/Header/index.jsx

```diff
@@ -1,4 +1,28 @@
+import { BrowserRouter as Router, NavLink } from 'react-router-dom';
+
 function Header() {
-  return <>Ceci est mon Header</>;
+  return (
+    <Router>
```

Comment:
Your `<Router>` should be in the root `index.js` file
(Right now, **only** your Header benefits from the Router, not the whole app)